### ISSUE TITLE
Ansible templating of swift.conf breaks MR recon & maas checks.

### ISSUE THREAD

**Issue by timolow** [rcbops/rpc-openstack#1112]
##### What version (commit) do you encounter this issue on?
12.0.13
##### What type of issue is this (bug, feature, documentation, regression)?
bug
##### What is the expected behavior?
swift.conf should match throughout the environment.  swift-recon --md5 should pass without errors.
##### What is actually happening?
ansible template in adding "# Ansible managed:" at the top thus causing:
```
`(swift-12.0.13)root@XXXXX-swiftproxy001_swift_proxy_container-ACBDEFG:~# swift-recon --md5
===============================================================================
--> Starting reconnaissance on 42 hosts
===============================================================================
[2016-06-02 09:29:58] Checking ring md5sums
42/42 hosts matched, 0 error[s] while checking hosts.
===============================================================================
[2016-06-02 09:29:58] Checking swift.conf md5sum
!! http://172.28.244.107:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.100:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.114:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.116:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.106:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.115:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.104:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.110:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.117:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.105:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.103:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.111:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.118:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.109:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.102:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.112:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.119:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.120:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.108:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.101:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
!! http://172.28.244.113:6000/recon/swiftconfmd5 (c1e92abe9bef64173049c0ea87267898) doesn't match on disk md5sum
21/42 hosts matched, 0 error[s] while checking hosts.
==============================================================================
```

 
##### How can we reproduce this issue?
Install MR swift and run recon.
##### How can we verify your issue is addressed?
swift-recon should pass all checks.
##### If this is a large issue, what can be done immediately to alleviate the issue while working on a longer-term solution?
remove template info from swift.conf
##### What is your evaluation on the importance of this issue, and why?
high
###### Remove branches (lines) you do not wish to target.
- [x] Master
- [x] liberty-12.2
- [x] liberty-12.1
- [x] kilo

**Comment by git-harry** [comment]
I have not attempted to reproduce this issue. From reviewing the report the issue appear legitimate. Removing the ansible_managed line from the template swift.conf.j2 requires a change upstream. An Ansible one-liner to modify swift.conf should be sufficient to work around the issue and so based on that and the limited impact of this issue I think a prio-2 is appropriate.

I have logged a bug [1] upstream with the details from this issue.

[1] https://bugs.launchpad.net/openstack-ansible/+bug/1589532

**Comment by mancdaz** [comment]
Upstream liberty: https://review.openstack.org/#/c/326374/
Upstream master (newton): https://review.openstack.org/#/c/326369/

Once upstream master has merged we can cherry pick that to upstream stable/mitaka

**Comment by mancdaz** [comment]
The upstream fixes will be included with the next sha bump for the OSA submodule

**Comment by evrardjp** [comment]
FYI the upstream mitaka merged.

**Comment by mancdaz** [comment]
SHA bump and upstream fix has been merged into liberty-12.2

**Comment by git-harry** [comment]
This has been addressed in master by 7c967a5127a78e992dff2fbe0bfbbcb2b71a3e35

**Comment by git-harry** [comment]
This still isn't fixed in Kilo
